tools/integration_tests/read_cache/helpers_test.go

```diff
@@ -175,9 +208,9 @@ func readFileAndValidateCacheWithGCS(ctx context.Context, storageClient *storage
 		// Validate cache size within limit.
 		validateCacheSizeWithinLimit(cacheCapacityInMB, t)
 	}
-	// Validate content read via gcsfuse with gcs.
-	client.ValidateObjectContentsFromGCS(ctx, storageClient, testDirName, filename,
-		expectedOutcome.content, t)
+	// Validate CRC32 of content read via gcsfuse with CRC32 value on gcs.
+	gotCRC32Value := crc32.Checksum([]byte(expectedOutcome.content), crc32cTable)
```

Comment:
would suggest to keep the way of calculating CRC same, so you can create a function which takes reader type/bytes slice and gives crc32 value and use that func here and above